test/gurka/test_recost.cc

```diff
@@ -0,0 +1,309 @@
+#include "gurka.h"
+#include "sif/util.h"
+#include <gtest/gtest.h>
+
+using namespace valhalla;
+
+const std::unordered_map<std::string, std::string> build_config{{"mjolnir.shortcuts", "false"}};
+
+TEST(recosting, all_algorithms) {
```

Comment:
lots of different tests in here. first one is a test of with and without time and almost all pairs of nodes in the map. making sure that the path and the recost numbers are exactly the same